torch/_functorch/_aot_autograd/subclass_utils.py

```diff
@@ -162,17 +162,20 @@ def wrap_tensor_subclasses(
     subclass_metas: List[Union[int, SubclassCreationMeta]],
     num_fw_outs_saved_for_bw: Optional[int] = None,
     is_runtime: bool = False,
+    num_tokens: int = 0,
```

Comment:
hmm. One thing I don't love about this solution is that the knowledge of effect tokens is leaking into the subclass desugaring logic. Ideally, they could be kept completely separate.

I think landing this is ok, but first I would like to better understand if there are any other alternatives. For example, I think what I would have expected is that the final function we trace out looks something like this:
```
def functionalized_flattened_tensor_fn(input_tokens, *plain_inputs):
    subclass_inputs = wrap_tensor_subclasses(plain_inputs)
    subclass_outs = original_f(*subclass_inputs)
    plain_outs = unwrap_tensor_sublcasses(subclass_outs)
    output_tokens = FunctionalTensorMode.get_output_tokens
    return *output_tokens, *plain_outs
```

(the above is pseudocode, but it doesn't feel fundamentally required for the extra "token" tensors to need to be passed into `wrap_tensor_subclasses`)

let me know if that makes sense, happy to chat more